### PR TITLE
Add documentation for Snowplow Media plugin for the JavaScript tracker

### PR DESCRIPTION
**Pull request by matus-tomlein**
This PR documents the new media tracking APIs in the Web.

Since the behaviour and APIs are similar for the JS and mobile trackers, I used a reusable component to have a single documentation for all of them. The component is then used in different places:

* plugin page for the JavaScript tracker
* plugin page for the browser tracker
* in future PR when we release the mobile media APIs: media tracking page for the mobile trackers

### CODE REVIEW

File: docs/reusable/media/_index.md [hunk 1]
```diff
@@ -0,0 +1,1064 @@
...
+</CodeBlock>)}</>
+
+## Tracking media events
+
+Having started a media tracking instance, you can use it to track media events as you receive them from the media player.
+
+Typically, you would subscribe to notifications from the media player (e.g., user clicks play, volume changes, content is buffering) with callbacks that would track the Snowplow events.
+For an example, see the code that subscribes for events from an HTML5 media player [here](https://github.com/snowplow-incubator/snowplow-javascript-tracker-examples/blob/issue/media_plugin_example/react/src/components/video.jsx).
+
+:::warning
```
Comment (greg-el) [inline docs/reusable/media/_index.md:405]: This can probably be removed now right?

File: docs/reusable/media/_index.md [hunk 2]
```diff
@@ -0,0 +1,1064 @@
...
+`iglu:com.snowplowanalytics.snowplow.media/quality_change_event/jsonschema/1-0-0`.
+
+##### Error
+
+Tracks a media player error event tracked when the resource could not be loaded due to an error.
+
+The event schema has the following properties:
+
+| Request Key | Required | Type/Format | Description |
+| --- | --- | --- | --- |
```
Comment (greg-el) [inline docs/reusable/media/_index.md:1018]: This should include `errorName` (and the examples below too)

**Review by greg-el — APPROVED** [review]
Looks good! A couple of comments, but once those have been addressed it should be good to go!

File: docs/reusable/media/_index.md (hunk 1 above)
Comment (matus-tomlein) [reply docs/reusable/media/_index.md]: Good catch! Removed.

File: docs/reusable/media/_index.md (hunk 2 above)
Comment (matus-tomlein) [reply docs/reusable/media/_index.md:1013]: Thanks, added!

**Pull request merged**